ui/app/components/secret-create-or-update.js

```diff
@@ -0,0 +1,249 @@
+/**
```

Comment:
these functions all came from `secret-edit`. I broke out that component into four components. 
1. secret-edit (handles creation of secretData and some permission things.
2. secretCreateOrUpdate, this component that handles all functions responsible for creating a new secret or editing one.
3. SecretEditToolbar, handles all functions with the toolbar.
4. SecretFormShow, handles the show view

Reply:
Also, secret-edit was an old component, but these new components are glimmer. Thus, I "glimmerized" these functions.